### PR TITLE
[SYCL-PTX] Add intermediate layer to libclc to ease type management

### PR DESCRIPTION
**Pull request by Naghasan**
This patch introduce a "core" layer, with function starting by `__clc_`, to implement builtin functions common to opencl and SPIR-V.

Some SPIR-V builtins are prefixed by a `s` or `u` to describe the type of the input. From the implementation point of view this serves no purpose (as it is carried by the type) and leads to a more complex usage of the builtins.
SPIR-V conversion function are expressed using various names, conversion implementation are now implemented as `__clc_convert_<desc>` and the opencl and SPIR-V convert function are implemented in terms of  `__clc_convert_*` functions

Note: this patch introduce a generated header (`spirv_builtins.h`), the generator will be upstreamed separately.

### CODE REVIEW

File: libclc/cmake/modules/AddLibclc.cmake [hunk 1]
```diff
@@ -42,8 +42,8 @@ endfunction(add_libclc_alias alias target)
 macro(add_libclc_builtin_set arch_suffix)
   cmake_parse_arguments(ARG
     ""
-    "TRIPLE;TARGET_ENV;LIB_DEP;GENERATE_TARGET;PARENT_TARGET"
-    "FILES;ALIASES;COMPILE_OPT"
+    "TRIPLE;TARGET_ENV;LIB_DEP;PARENT_TARGET"
+    "FILES;ALIASES;GENERATE_TARGET;COMPILE_OPT"
```
Comment (bader) [inline libclc/cmake/modules/AddLibclc.cmake:46]: Interesting... is there any difference between old and new code?

File: libclc/generic/libspirv/integer/upsample.cl [hunk 2]
```diff
@@ -1,46 +1,40 @@
-//===----------------------------------------------------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
+#include <clc/clc.h>
```
Comment (bader) [inline libclc/generic/libspirv/integer/upsample.cl:1]: Shouldn't we keep the header with license information?

File: libclc/cmake/modules/AddLibclc.cmake (hunk 1 above)
Comment (Naghasan) [reply libclc/cmake/modules/AddLibclc.cmake:46]: yes, the first line model single value variable, the second list of variable. Welcome to the wonder of cmake :)

`GENERATE_TARGET` is now a list of targets, in sync with this change https://github.com/intel/llvm/pull/1712/files#diff-b25aa3b585676dbf10fc08d937f595e0R240

File: libclc/generic/libspirv/integer/upsample.cl (hunk 2 above)
Comment (Naghasan) [reply libclc/generic/libspirv/integer/upsample.cl:1]: oh damn, yes. I'll fix that

Comment (Naghasan) [reply libclc/generic/libspirv/integer/upsample.cl:9]: fixed

File: libclc/cmake/modules/AddLibclc.cmake (hunk 1 above)
Comment (bader) [reply libclc/cmake/modules/AddLibclc.cmake:46]: Ah... My bad. I missed that we have two arguments here. I thought it's a single list...

**Pull request merged**